app/logic/Mail/SQL/SQLEMail.ts

```diff
@@ -223,19 +223,89 @@ export class SQLEMail {
   static async saveMultiple(emails: Collection<EMail>) {
     let lock = await this.transactionLock.lock();
     try {
-      await (await getDatabase()).run(sql`BEGIN TRANSACTION`);
-      for (let email of emails) {
-        if (!email.subject) {
-          continue;
+      // We need to save all persons before the email transaction
+      // otherwise it will not appear in the database
+      let db = await getDatabase();
+      await db.executeTransaction(() => {
+        for (let email of emails) {
+          if (!email.subject) {
+            continue;
+          }
+          this.saveMainProperties(db, email);
+          this.savePersons(db, email);
         }
-        await this.save(email);
-      }
-      await (await getDatabase()).run(sql`END TRANSACTION`);
+      });
     } finally {
       lock.release();
     }
   }
 
+  protected static saveMainProperties(db: Database, email: EMail) {
+    let contact = email.contact as PersonUID;
+    db.run(sql`
```

Comment:
This is an async function that you call without `await`. *All* calls to the database are always going to the backend, and the communication between frontend and backend is happening over WebSockets and therefore always async. There's no way to make sync calls to the DB from the frontend.